### PR TITLE
fix: download correct rc or ga release

### PR DESCRIPTION
**Pull request by caspahouzer**
This PR fixes the download of the correct RC or GA version of the SDK.

To test, just try
`ti sdk install latest`
`ti sdk install 11.0.0.RC`
`ti sdk install 10.1.1.GA`

If you have already installed a version use `--force`

Big kudos to **Bruno**, who make this PR possible

### CODE REVIEW

**Review by ecarruda — APPROVED** [review]

File: lib/commands/sdk.js [hunk 1]
```diff
@@ -1444,18 +1444,21 @@ exports.getReleases = async function getReleases(config, os) {
 		// Paginate through the releases on the repo, and determine which of the assets
 		// is correct for the platform we're running on
 		const releaseData = await octokit.paginate(paginateData);
-		const osRegex = /mobilesdk-\d+\.\d+\.\d+\.GA-(\w+)\.zip/;
+		const osRegex = /mobilesdk-\d+\.\d+\.\d+\.([A-Z)]{2})-(\w+)\.zip/;
 		for (let { assets, name } of releaseData) {
 			const release = assets.find(asset => {
-				const [ , assetOS ] = osRegex.exec(asset.name);
+				const [ , versionType, assetOS ] = osRegex.exec(asset.name);
```
Comment (ecarruda) [inline lib/commands/sdk.js:1450]:
```suggestion
				const [ , , assetOS ] = osRegex.exec(asset.name);
```
versionType is not used.

File: lib/commands/sdk.js [hunk 2]
```diff
@@ -1444,18 +1444,21 @@ exports.getReleases = async function getReleases(config, os) {
...
 				}
 				return false;
 			});
 
 			if (release) {
-				name = name.includes('GA') ? name : `${name}.GA`;
+				const analyseVersionRegExp = /(\d+\.\d+\.\d+)(\.)*([A-Z)]{2})*/;
+				const [ , version, dot, versionType ] = analyseVersionRegExp.exec(name);
+				// Fallback in GA version for backward compatibility
+				name = !versionType || !dot ? `${version}.GA` : `${version}.${versionType}`;
```
Comment (ecarruda) [inline lib/commands/sdk.js:1461]:
```suggestion
				if (!version || !versionType || !dot) {
					name = name.includes('GA') ? name : `${name}.GA`;
				} else {
					name = `${version}.${versionType}`;
				}
```

Why not let as fallback the old code propose?

**Comment by ecarruda** [comment]
I created this issue and was going to publish my PR when i saw your PR linking my issue, just some ideas about what I did on my branch.

File: lib/commands/sdk.js (hunk 2 above)
Comment (ecarruda) [reply lib/commands/sdk.js:1461]: This is just an idea because RC is always o the name (at least until now).

**Review by ecarruda — CHANGES_REQUESTED** [review]

File: lib/commands/sdk.js (hunk 1 above)
Comment (ecarruda) [inline lib/commands/sdk.js:1450]:
```suggestion
				const [ , , assetOS ] = osRegex.exec(asset.name);
```

versionType is not used

**Review by ecarruda — CHANGES_REQUESTED** [review]

**Comment by caspahouzer** [comment]
```
ti sdk install 11.0.0.RC
Titanium SDK 11.0.0.RC is already installed!

ti sdk install latest
You're up-to-date. Version 10.1.1.GA is currently the newest version available.
```

**Comment by caspahouzer** [comment]
Closing this PR in favour of https://github.com/tidev/titanium/pull/583

**Pull request closed without merging**